src/Parser/Core/Modules/StatTracker.js

```diff
@@ -0,0 +1,257 @@
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import SPECS from 'common/SPECS';
+import { calculateSecondaryStatDefault } from 'common/stats';
+import { formatMilliseconds } from 'common/format';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+const debug = false;
+
+// TODO: add stam, speed, avoidance, leech
+// TODO: stat constants somewhere else? they're largely copied from combatant
+class StatTracker extends Module {
```

Comment:
I believe this should be part of the Combatant class as soon as it's merged. If it's not we will soon find us with modules relying on this and it would become much harder to do. Migrating all existing usages of the properties of Combatant will be worth the trouble (and messy code) saved.

Reply:
My concern is that the functionality presented by this module only really works on the selected combatant, while the Combatant class is meant for all combatants.

Also, this functionality is implemented as a Module, and Combatant is not a module. Do you mean that this should be in Combatants? I'm not sure what you envision going where.

Reply:
Hmmm yes this might be annoying. I quite dislike the Entities/Entity setup we have right now.

I think this would have been decent as a SelectedCombatant : Combatant class, but it doesn't solve the event handler issue.